sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -1043,6 +1043,7 @@ public partial class HttpPipelineTransportOptions
     {
         public HttpPipelineTransportOptions() { }
         public System.Collections.Generic.IList<System.Security.Cryptography.X509Certificates.X509Certificate2> ClientCertificates { get { throw null; } }
+        public System.Func<Azure.Core.HttpMessage, bool>? ClientRedirectAuthorizationRemovalCallback { get { throw null; } }
```

Comment:
For some options, we've been adding static APIs to the relevant policies that put a value in the HttpMessage's property bag.  This makes it so we don't end up with a very large set of ClientOptions where many are for relatively uncommon scenarios.  For example, we did this when we added the static SetAllowAutoRedirect method on the RedirectPolicy.

Is that an approach we might consider using in this case?